sml_small/utils/common_utils.py

```diff
@@ -80,3 +82,61 @@ def is_number(value) -> bool:
         return False
 
     return True
+
+
+def convert_input_to_decimal(
+    keys: List[str],
+    args: List[float],
+) -> Dict[str, Decimal]:
+    """
+    convert_input_to_decimal This function will take key value pairs within a dict
+    the values will then be converted to decimal if possible.
+
+    :param keys: List of string names associated with the values in the list
+    :type keys: List[str]
+    :param args: Values to be converted to decimal
+    :type args: List[float]
+    :raises ValueError: Error string raised in the event the keys do not
+    have arguments or arguments do not have keys
+    ...
+    :return: key names and their values returned as decimals if appropriate.
+    :rtype: Dict[str, Decimal]
+    """
+    try:
+        if len(keys) != len(args):
+            raise ValueError("Number of keys needs to match the number of arguments")
+
+        decimal_values = {}
+
+        # Using zip to iterate through the keys and arguments simultaneously
+        for key, arg in zip(keys, args):
+
+            if type(arg) == Decimal:
+                decimal_values[key] = arg
+
+            elif type(arg) == list:
+
+                if arg == []:
+                    decimal_values[key] = arg
+
+                else:
+                    for i in range(len(arg)):
+                        if type(arg[i]) == Decimal:
+                            arg[i] = arg[i]
```

Comment:
This doesn't look correct - should be decimal_values[key] = arg ?

Reply:
Hi Steve, yes good point there it would not set the value otherwise. I have changed it.